tracing: misc cleanups

andreimatei opened the pull request:
See individual commits.

andreimatei committed:
tracing: remove stale TODO

This TODO implied that orphaned spans (spans whose parent is
Finish()ing) are not made part of the active spans registry. But they
are, just above the TODO.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72727)
<!-- Reviewable:end -->

andreimatei committed:
tracing: remove unused method

The span.ResetRecording() method was fairly unused. It's also in my way
because it clears the children of a span without properly making them
part of the active spans registry (like for example parent.Finish()
would do). This patch gets rid of it.

Release note: None

abarganier reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72727#-:-ModSat79AdyYN-D3SWP:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72727#-:-MonbN8d36yAgOpafVjr:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->
